specification/app/resource-manager/Microsoft.App/preview/2023-05-01-preview/Builds.json

```diff
@@ -0,0 +1,529 @@
+        "responses": {
+          "200": {
+            "description": "Resource deleted successfully."
+          },
+          "202": {
+            "description": "Resource deletion accepted.",
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "Resource deleted successfully."
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Builds_Delete_0": {
+            "$ref": "./examples/Builds_Delete.json"
+          }
+        },
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-long-running-operation": true
+      }
+    }
+  },
+  "definitions": {
+    "BuildConfiguration": {
+      "type": "object",
+      "properties": {
+        "baseOs": {
+          "type": "string",
+          "description": "Base OS used to build and run the app."
+        },
+        "platform": {
+          "type": "string",
+          "description": "Platform to be used to build and run the app."
+        },
+        "platformVersion": {
+          "type": "string",
+          "description": "Platform version to be used to build and run the app."
+        },
+        "environmentVariables": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EnvironmentVariable"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "description": "List of environment variables to be passed to the build and application runtime."
+        },
+        "preBuildSteps": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PreBuildStep"
+          },
+          "x-ms-identifiers": [],
+          "description": "List of steps to perform before the build."
+        }
+      },
+      "description": "Configuration of the build."
+    },
+    "BuildProperties": {
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "Build provisioning state.",
+          "readOnly": true
+        },
+        "buildStatus": {
+          "$ref": "#/definitions/BuildStatus",
+          "description": "Status of the build once it has been provisioned.",
+          "readOnly": true
+        },
+        "destinationContainerRegistry": {
+          "$ref": "#/definitions/ContainerRegistryWithCustomImage",
+          "description": "Container registry that the final image will be uploaded to."
+        },
+        "configuration": {
+          "$ref": "#/definitions/BuildConfiguration",
+          "description": "Configuration of the build."
+        },
+        "uploadEndpoint": {
```

Comment:
How does the user auth to this endpoint?

Reply:
The plan is that RBAC will happen during this Build creation call, and we can then pass back a SAS token with short lifetime in this `uploadEndpoint`